modules/database/src/std/rec/longoutRecord.c

```diff
@@ -421,10 +435,61 @@ static long writeValue(longoutRecord *prec)
 
 static void convert(longoutRecord *prec, epicsInt32 value)
 {
-        /* check drive limits */
+    /* check drive limits */
     if(prec->drvh > prec->drvl) {
         if (value > prec->drvh) value = prec->drvh;
         else if (value < prec->drvl) value = prec->drvl;
     }
     prec->val = value;
 }
+
+/* Evaluate OOPT field to perform the write operation */
+static long conditional_write(longoutRecord *prec)
+{
+    struct longoutdset *pdset = (struct longoutdset *) prec->dset;
+    long status = 0;
+    int doDevSupWrite = 0;
+
+    switch (prec->oopt) 
+    {
+    case longoutOOPT_On_Change:
+        /* Forces a write op if a change in the OUT field is detected */
+        if ((prec->ooch == menuYesNoYES) && (prec->outpvt == OUT_LINK_CHANGED)) {
+            doDevSupWrite = 1;
+        } else {
+            /* Only write if value is different from the previous one */ 
+            doDevSupWrite = (prec->val != prec->pval);
+        }
+        break;
+
+    case longoutOOPT_Every_Time:
+        doDevSupWrite = 1;
+        break;
+
+    case longoutOOPT_When_Zero:
+        doDevSupWrite = (prec->val == 0);
+        break;
+
+    case longoutOOPT_When_Non_zero:
+        doDevSupWrite = (prec->val != 0);
+        break;
+
+    case longoutOOPT_Transition_To_Zero:
+        doDevSupWrite = ((prec->val == 0)&&(prec->pval != 0));
+        break;
+
+    case longoutOOPT_Transition_To_Non_zero:
+        doDevSupWrite = ((prec->val != 0)&&(prec->pval == 0));      
+        break;
+
+    default:
+        break;
+    }
+
+    if (doDevSupWrite)
+        status = pdset->write_longout(prec);
+
+    prec->pval = prec->val;
+    prec->outpvt = OUT_LINK_UNCHANGED; /* reset status of OUT link */
+    return status;
+}
```

Comment:
Can you add a newline to the end of this file please. Posix requires all text files to end with a newline and all the Unix editors would include one automatically; I think it was editors that came from the Windows world that started leaving it off, which can cause some surprising bad effects.